### ISSUE TITLE
Not a Function/Unexpected Token

### ISSUE THREAD

**Issue by resech** [CapacitorSet/box-js#32]
Hello,

Having a few issues with this sample: [jssample.zip](https://github.com/CapacitorSet/box-js/files/1247406/jssample.zip) (password:infected). When I run it with the default options I see:

```Using a 10 seconds timeout, pass --timeout to specify another timeout in seconds
[debug] Analysis launched: ["/usr/bin/nodejs","/opt/box-js/analyze","jssample.js","jssample.js.results/","--loglevel","debug","--timeout=10"]
[verb] Analyzing jssample.js
[debug] Using detected encoding
[debug] jschardet (v1.5.1) detected encoding ascii
[verb] Rewriting code...
[verb]     Replacing `function A.prototype.B()` (use --no-rewrite-prototype to skip)...
[verb]     Rewriting typeof calls (use --no-typeof-rewrite to skip)...
[verb]     Rewriting eval calls (use --no-eval-rewrite to skip)...
[verb]     Rewriting try/catch statements (use --no-catch-rewrite to skip)...
[verb] Rewritten successfully.
[verb] Code saved to 87e6db80-a18f-4365-9f46-fcd3d3b2aed2.js
[debug] Analyzing with vm2 v3.4.6

/opt/box-js/node_modules/vm2/lib/main.js:213
			throw this._internal.Decontextify.value(e);
			^
TypeError: "457870616EcTxqapTV6445zo hXdZ6E76Gz q69726Fwpt6E6DtqNp65acSG ywzpnq6EttwvVVfLaW7453afqskt7472npeRwg696Esmvc p6773".w is not a function
    at vm.js:42:130
    at ContextifyScript.Script.runInContext (vm.js:35:29)
    at VM.run (/opt/box-js/node_modules/vm2/lib/main.js:207:72)
    at Object.<anonymous> (/opt/box-js/analyze.js:380:5)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.runMain (module.js:604:10)

 * If the error is about a weird "Unknown ActiveXObject", try --no-kill.
 * If the error is about a legitimate "Unknown ActiveXObject", report a bug at https://github.com/CapacitorSet/box-js/issues/ .
```
If I try it with ```--no-rewrite``` I get the following:

```Using a 10 seconds timeout, pass --timeout to specify another timeout in seconds
[debug] Analysis launched: ["/usr/bin/nodejs","/opt/box-js/analyze","jssample.js","jssample.js.1.results/","--loglevel","debug","--no-rewrite","--timeout=10"]
[verb] Analyzing jssample.js
[debug] Using detected encoding
[debug] jschardet (v1.5.1) detected encoding ascii
[verb] Code saved to ee238066-3abf-41cd-a922-477f3c641d3b.js
[debug] Analyzing with vm2 v3.4.6

/opt/box-js/node_modules/vm2/lib/main.js:213
			throw this._internal.Decontextify.value(e);
			^
SyntaxError: Unexpected token .
    at VMScript.compile (/opt/box-js/node_modules/vm2/lib/main.js:74:20)
    at VM.run (/opt/box-js/node_modules/vm2/lib/main.js:207:52)
    at Object.<anonymous> (/opt/box-js/analyze.js:380:5)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.runMain (module.js:604:10)
    at run (bootstrap_node.js:389:7)

 * If the error is about a weird "Unknown ActiveXObject", try --no-kill.
 * If the error is about a legitimate "Unknown ActiveXObject", report a bug at https://github.com/CapacitorSet/box-js/issues/ .
```
I didn't see anything in the other issues that seemed applicable to this one but I'd love to know what's going on.

**Comment by resech** [comment]
I just noticed #31, seems like it may resolve this issue once it's ready.

**Comment by CapacitorSet** [comment]
Indeed, the error you reported is due to the function hoisting mentioned in #31, but if you fix that manually the script still won't work due to discrepancies between Node.js' and WScript's handling of `new Date()`. I'll investigate the issue.

**Comment by CapacitorSet** [comment]
There were two issues, one with the Date discrepancy (fixed in e73add7917a0ebbd8904e94a8cfa770efc8e89ee), and one with the missing implementation of XMLHTTP.statusText (fixed in 8824f735b45a6931abcc21075eadae160ac18b89). Thanks for your contribution! Now let's wait for the linked PR to be ready for merging.
